src/main/java/org/openstreetmap/atlas/checks/validation/linear/edges/SingleSegmentMotorwayCheck.java

```diff
@@ -73,8 +78,14 @@ public boolean validCheckForObject(final AtlasObject object)
                 .noneMatch(this::isMotorwayNotRoundabout))
         {
             this.markAsFlagged(edge.getOsmIdentifier());
-            return Optional.of(this.createFlag(new OsmWayWalker(edge).collectEdges(),
-                    this.getLocalizedInstruction(0, edge.getOsmIdentifier())));
+            final CheckFlag flag = this.createFlag(new OsmWayWalker(edge).collectEdges(),
+                    this.getLocalizedInstruction(0, edge.getOsmIdentifier()));
+            final Optional<String> fixSuggestion = this.getFixSuggestion(edge);
+            return fixSuggestion.map(s -> flag.addFixSuggestion(FeatureChange.add(
```

Comment:
Please use a more descriptive variable name than `s`.

Reply:
Renamed to suggestion.